### PR TITLE
new(falco): add gVisor support

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

> If contributing rules or changes to rules, please make sure to also uncomment one of the following line:

> /kind rule-update

> /kind rule-create

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area engine

> /area rules

> /area tests

> /area proposals

**What this PR does / why we need it**:

This PR adds support for the Falco CLI to enable syscall data collection from gVisor, as supported by libsinsp and libscap. Before this can be tested it needs the libs version to be advanced to the appropriate commit in Falco.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

See inline comments.

**Does this PR introduce a user-facing change?**:

```release-note
new(falco): add gVisor support
```

### CODE REVIEW

File: userspace/falco/application.h
Comment (LucaGuerra) [inline userspace/falco/application.h:198]: Note for the reviewer: the astute reader will have noticed that the boolean method `is_gvisor()` does not exist yet in falco libs, so I'm going to add it. I think it's the clean way to implement this check but if maintainers and contributors have better ideas I can change the implementation.

Comment (LucaGuerra) [reply userspace/falco/application.h:198]: Now it is implemented.

**Comment by FedeDP** [comment]
I think this should be rebased on master to fix integration tests!

**Comment by LucaGuerra** [comment]
* PR is rebased
* It now requires https://github.com/falcosecurity/libs/pull/429 because the interface changed.

**Comment by LucaGuerra** [comment]
Now it's rebased on top of libs that expose all the necessary functionality! :tada:

File: userspace/falco/app_cmdline_options.cpp
Comment (FedeDP) [reply userspace/falco/app_cmdline_options.cpp:165]: This LGTM! I am waiting for this part and we are ready to go IMO :)

Comment (leogr) [reply userspace/falco/app_cmdline_options.cpp:165]: Use the `MINIMAL_BUILD` pattern please :point_down: 
https://github.com/falcosecurity/falco/blob/master/userspace/falco/app_cmdline_options.cpp#L166

Options must be present only when Falco has been built with the gVisor support.

Furthermore, note that the gVisor support must be disabled when `MINIMAL_BUILD` is ON.

File: cmake/modules/falcosecurity-libs.cmake
Comment (leogr) [inline cmake/modules/falcosecurity-libs.cmake:30]: I'd suggest tagging a version in libs before merging this PR.

@FedeDP wdyt?

**Review by leogr — COMMENTED** [review]
This is awesome! :star_struck: 

Just left two comments (see below)

File: userspace/falco/app_actions/init_falco_engine.cpp
Comment (jasondellaluce) [inline userspace/falco/app_actions/init_falco_engine.cpp:48]:
```suggestion
		output_format = "container=%container.id pid=%proc.vpid tid=%thread.vtid";
```

File: cmake/modules/falcosecurity-libs.cmake
Comment (FedeDP) [reply cmake/modules/falcosecurity-libs.cmake:30]: Makes sense! :+1:

**Review by jasondellaluce — COMMENTED** [review]
Left you some comments Luca! Mostly nitpicking

File: userspace/falco/app_cmdline_options.cpp
Comment (LucaGuerra) [reply userspace/falco/app_cmdline_options.cpp:165]: I'm following that same pattern, reverting the PR to wip until I'm relatively happy with the result

File: userspace/falco/application.h
Comment (LucaGuerra) [reply userspace/falco/application.h]: Good catch! I'm not completely convinced of both but now I see that the one just above it does exactly what you describe. I updated that to reflect this.

**Comment by LucaGuerra** [comment]
@leogr @jasondellaluce @FedeDP addressed review comments.
I also checked that it builds without gvisor support and in that case the CLI options related to it are not displayed at all.

**Comment by FedeDP** [comment]
>  I also checked that it builds without gvisor support and in that case the CLI options related to it are not displayed at all.

Did you also check that gvisor is actually not built inside libscap? I mean, that eg: you don't see the protobuf dep in libscap.

**Comment by FedeDP** [comment]
IMO this LGTM! I think that we can wait for https://github.com/falcosecurity/libs/pull/433 and https://github.com/falcosecurity/libs/pull/416 in libs and tag a new libs/driver release that can possibly be the final for the release.
WDYT? 
Do you want this merged before those 2 PRs?

**Comment by LucaGuerra** [comment]
@FedeDP  The PRs you mentioned are in and the versions have been updated in master. This can now be merged on top of master with the current RC libs versions. I just rebased the branch accordingly.

**Review by FedeDP — APPROVED** [review]
/approve

**Review by jasondellaluce — APPROVED** [review]

**Pull request merged**
